### PR TITLE
Add test that diagonal concat with mismatching schemas raises

### PR DESCRIPTION
**Pull request by wence-**
## Description
Arguably this should be determined during query optimization by polars, but for now it is raised late during compute, so we must validate on our side.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf_polars/cudf_polars/dsl/ir.py [hunk 1]
```diff
@@ -933,10 +933,10 @@ class Union(IR):
     """Optional slice to apply after concatenation."""
 
     def __post_init__(self) -> None:
-        """Validated preconditions."""
+        """Validate preconditions."""
         schema = self.dfs[0].schema
         if not all(s.schema == schema for s in self.dfs[1:]):
-            raise ValueError("Schema mismatch")
```
Comment (lithomas1) [inline python/cudf_polars/cudf_polars/dsl/ir.py:939]: Why the change here?

Is the idea to give user the polars version of the error message.

**Review by lithomas1 — APPROVED** [review]
Looks good!

Left one question for my understanding.

File: python/cudf_polars/cudf_polars/dsl/ir.py (hunk 1 above)
Comment (wence-) [reply python/cudf_polars/cudf_polars/dsl/ir.py:939]: The promise of `translate_ir` is that it will fail, with `NotImplementedError`, if we can't do the thing. This `ValueError` exception broke that promise. Arguably this is just a stop-gap for polars not validating the schema early. I'm not trying to match polars exceptions one-for-one.
